packages/components/label/src/label.tsx

```diff
@@ -0,0 +1,77 @@
+import type { MessageDescriptor } from 'react-intl';
+
+import React from 'react';
+import invariant from 'tiny-invariant';
+import Text from '@commercetools-uikit/text';
+import RequiredIndicator from './required-indicator';
+
+type TBasicTextProps = {
+  intlMessage?: MessageDescriptor;
+  children?: React.ReactNode;
+};
+type TLabelProps = {
+  /**
+   * Use this to set the relationships between the label and a non-form element.
+   * For example:
+   *   <label id="foo">
+   *   <div aria-labelledby="foo">
+   */
+  id?: string;
+  /**
+   * Use this to set the relationships between the label and a form element.
+   * For example:
+   *   <label for="foo">
+   *   <input id="foo">
+   *
+   *   <label id="foo">
+   *   <input aria-labelledby="foo">
+   */
+  htmlFor?: string;
+  isBold?: boolean;
+  isRequiredIndicatorVisible?: boolean;
+  tone?: 'primary' | 'inverted';
+} & TBasicTextProps;
+
+const Label = (props: TLabelProps) => {
+  invariant(
+    !(Boolean(props.id) && Boolean(props.htmlFor)),
+    `ui-kit/Label: provide only the "id" or the "htmlFor" properties, not both.`
+  );
+
+  const hasContent =
+    props.intlMessage || Boolean(React.Children.count(props.children));
+
+  invariant(
+    hasContent,
+    [
+      'Warning: Failed prop type:',
+      `The prop \`intlMessage\` is marked as required in \`Label\``,
+      'but its value is `undefined`',
+    ].join(' ')
+  );
+  invariant(
+    hasContent,
+    [
+      'Warning: Failed prop type:',
+      `The prop \`children\` is marked as required in \`Label\``,
+      'but its value is `undefined`',
+    ].join(' ')
+  );
+
+  return (
+    <label id={props.id} htmlFor={props.htmlFor}>
+      <Text.Body
+        tone={props.tone}
+        isBold={props.isBold}
+        intlMessage={props.intlMessage}
+      >
+        {props.children}
+        {props.isRequiredIndicatorVisible && <RequiredIndicator />}
+      </Text.Body>
```

Comment:
Looking at this, if you pass `intlMessage`, the Text children are not rendered, which means that the required indicator does not get rendered.

Can you check that please? It might be that we have always had this "issue" ...

Reply:
that is a great catch! will revert to previous solution where we render `intlMessage ? <FormattedMessage /> : children`